### ISSUE TITLE
Overflow of softmax function

### ISSUE THREAD

**Issue by Oyamatsumi** [yangpeiren/NNPraktikum#3]
As feed_forward function proceeds, some input data will cause exp(a) to be too small for float64, which ends up "nan" values that corrupt the network.

**Comment by Oyamatsumi** [comment]
It seems to have something to do with our implemented loss function bce. As I use the given sse, there are no more overflows without a scaler.
